src/librustc_resolve/lib.rs

```diff
@@ -2194,7 +2177,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
     // user and one 'x' came from the macro.
     fn binding_mode_map(&mut self, pat: &Pat) -> BindingMap {
         let mut result = HashMap::new();
-        pat_bindings(&self.def_map, pat, |binding_mode, _id, sp, path1| {
+        pat_bindings(&self.def_map,
+                     pat,
+                     |binding_mode, _id, sp, path1| {
```

Comment:
Hmm, I agree that the `walk_pat` formatting does not look great, and I imagine the same will happen for some uses of `map_or`. However, I do think in many cases (mostly without closures) having a sub-expression split vertically, but not stacking all args looks very odd. I wonder if the exception should be for closures? Or for the last argument maybe?